File: frontend/src/types/match.ts
export interface JobMatch {
  job_id: string
  title: string
  company: string
  location: string
  remote: string
  match_score: number
  reasoning: string
  highlight_bullet: string
}

export type StreamEventType = 'status' | 'match' | 'done' | 'error'

export interface StatusEvent {
  stage: 'parsing' | 'embedding' | 'retrieving' | 'ranking'
}

export interface DoneEvent {
  total: number
  duration_ms: number
  cache_hit?: boolean
}

export interface ErrorEvent {
  message: string
}

File: backend/app/schemas/match.py
from typing import Literal

from pydantic import BaseModel, Field


class JobMatch(BaseModel):
    job_id: str
    title: str
    company: str
    location: str
    remote: str
    match_score: int = Field(ge=0, le=100)
    reasoning: str
    highlight_bullet: str


class StatusEvent(BaseModel):
    stage: Literal["parsing", "embedding", "retrieving", "ranking"]


class DoneEvent(BaseModel):
    total: int
    duration_ms: int


class ErrorEvent(BaseModel):
    message: str


class MatchStreamEvent(BaseModel):
    event: Literal["status", "match", "done", "error"]
    data: StatusEvent | JobMatch | DoneEvent | ErrorEvent
